lib/Dialect/HW/HWOps.cpp

```diff
@@ -2495,58 +2498,57 @@ void ArrayGetOp::build(OpBuilder &builder, OperationState &result, Value input,
 // single uniform value for each element, just return that value regardless of
 // the index. If the array is constructed from a constant by a bitcast
 // operation, we can fold into a constant.
-OpFoldResult ArrayGetOp::fold(ArrayRef<Attribute> operands) {
-  if (operands[0]) {
-    auto arrayAttr = operands[0].cast<ArrayAttr>();
+OpFoldResult ArrayGetOp::fold(FoldAdaptor adaptor) {
+  auto inputCst = adaptor.getInput().dyn_cast_or_null<ArrayAttr>();
+  auto indexCst = adaptor.getIndex().dyn_cast_or_null<IntegerAttr>();
+
+  if (inputCst) {
     // Constant array index.
-    if (operands[1]) {
-      auto index = operands[1].cast<IntegerAttr>().getValue();
-      return arrayAttr[arrayAttr.size() - 1 - index.getZExtValue()];
+    if (indexCst) {
+      auto index = indexCst.getValue().getZExtValue();
```

Comment:
I wonder if this is missing a check that the index will fit in a 64 bit value. There is a similar coercion repeated below that does this:
```c++
  if (!indexCst || indexCst.getValue().getBitWidth() > 64)
    return {};
  uint64_t index = indexCst.getValue().getLimitedValue();
```
Maybe a follow up pr?